ext/reflection/php_reflection.c

```diff
@@ -3668,6 +3706,197 @@ ZEND_METHOD(reflection_method, setAccessible)
 }
 /* }}} */
 
+/* {{{ proto public void ReflectionClassConstant::__construct(mixed class, string name)
+   Constructor. Throws an Exception in case the given class constant does not exist */
+ZEND_METHOD(reflection_class_constant, __construct)
+{
+	zval *classname, *object, name, cname;
+	zend_string *constname;
+	reflection_object *intern;
+	zend_class_entry *ce;
+	zend_class_constant *constant = NULL;
+
+	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "zS", &classname, &constname) == FAILURE) {
+		return;
+	}
+
+	object = getThis();
+	intern = Z_REFLECTION_P(object);
+	if (intern == NULL) {
```

Comment:
This is not my code, but you are right. This check may be removed. I'll do
it.

On Mon, Dec 7, 2015 at 7:34 PM, Xinchen Hui <notifications@github.com>
wrote:

> In ext/reflection/php_reflection.c
> <https://github.com/php/php-src/pull/1662#discussion_r46842248>:
>
> > +   Constructor. Throws an Exception in case the given class constant does not exist */
> > +ZEND_METHOD(reflection_class_constant, __construct)
> > +{
> > +	zval *classname, *object, name, cname;
> > +	zend_string *constname;
> > +	reflection_object *intern;
> > +	zend_class_entry *ce;
> > +	zend_class_constant *constant = NULL;
> > +
> > +	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "zS", &classname, &constname) == FAILURE) {
> > +		return;
> > +	}
> > +
> > +	object = getThis();
> > +	intern = Z_REFLECTION_P(object);
> > +	if (intern == NULL) {
>
> How can it be NULL?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/1662/files#r46842248>.
>